bson/src/test/unit/org/bson/DocumentTest.java

```diff
@@ -79,6 +79,16 @@ public void toJsonShouldReturnEquivalent() {
                      document);
     }
 
+    // Test to ensure that toJson does not reorder _id field
+    @Test
+    public void toJsonShouldNotReorderIdField() {
+        // given
+        Document d = new Document().append("x", 1)
+                .append("y", Collections.singletonList("one"))
+                .append("_id", "1");
+        assertEquals(new DocumentCodec().decode(new JsonReader(d.toJson()), DecoderContext.builder().build()), d);
```

Comment:
Make this a more direct test by asserting equality on the actual JSON that is return by toJson

Reply:
I have updated the test.